src/graphql/course/service.ts

```diff
@@ -48,59 +48,44 @@ export class CourseService {
    * @returns a list of `Course`
    */
   public async coursesBy(pred: QueryInput): Promise<Course[]> {
-    let res: Course[];
-    if (pred.departmentCode?.length == 0) pred.departmentCode = undefined;
+    const deptCodeQuery = () => {
```

Comment:
Ternaries make these closures hard to read. Consider defining these functions using guard clauses, possibly outside this function, or adding comments